### PR TITLE
Polymorphic default

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Fixes issues in Polymorphic. Adds a `DefaultAs` generic prop 

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/polymorphic/src/Polymorphic/index.ts [hunk 1]
```diff
@@ -9,15 +8,3 @@ export type {
...
-  toBePolymorphic,
-});
-
-declare global {
-  namespace jest {
-    interface Matchers<R> {
-      toBePolymorphic(): R;
-    }
-  }
-}
```
Comment (TheSonOfThomp) [inline packages/polymorphic/src/Polymorphic/index.ts:23]: This causes issues when importing the component. We'll need to extend expect in testing lib https://jira.mongodb.org/browse/LG-2733

**Comment by brookescarlett** [comment]
I think we've moved too hard in the other direction... not seeing any TS errors anymore when misusing as/href

File: packages/polymorphic/src/Example/examples.spec.tsx [hunk 2]
```diff
@@ -59,6 +60,15 @@ describe('Polymorphic Higher-order Components', () => {
       expect(getByTestId('styled')).toHaveAttribute('href', 'mongodb.design');
       expect(getByTestId('styled')).toHaveStyle(`color: #ff69b4;`);
     });
+
+    test('Works with a default, and sets default props', () => {
```
Comment (brookescarlett) [inline packages/polymorphic/src/Example/examples.spec.tsx:64]: should probably test that as/href work as expected as well

File: packages/polymorphic/src/Example/Polymorphic.example.tsx [hunk 3]
```diff
@@ -48,16 +52,37 @@ export const ExamplePolymorphicWithRef = Polymorphic<ExampleProps>(
   'ExamplePolymorphicWithRef',
 );
 
-export const ExampleInferred = Polymorphic<
-  InferredPolymorphicProps<ExampleProps>
->(({ as, title, ...rest }) => {
+/**
+ * @example
+ */
+export const ExampleInferred = InferredPolymorphic<ExampleProps>(
```
Comment (brookescarlett) [inline packages/polymorphic/src/Example/Polymorphic.example.tsx:58]: does this need defaults? What happens if there are none?

Comment (TheSonOfThomp) [reply packages/polymorphic/src/Example/Polymorphic.example.tsx:58]: This is an example for usage without a default. 

`ExampleInferredDefaultButton` is the example for usage with a default

File: scripts/utils/tsDocParser.ts [hunk 4]
```diff
@@ -58,17 +58,47 @@ const compilerOptions: CompilerOptions = {
 
 const Parser = withCompilerOptions(compilerOptions, TSDocOptions);
 
+interface ParseFunctionOptions {
+  /**
```
Comment (brookescarlett) [inline scripts/utils/tsDocParser.ts:62]: are these changes related to poly default or should this be its own PR?

File: packages/polymorphic/README.md [hunk 5]
```diff
@@ -83,6 +83,24 @@ export const MyInferredComponent = Polymorphic<
 <MyInferredComponent href="mongodb.design" />; // renders as <a>
 ```
 
+#### With a default `as` value
```
Comment (brookescarlett) [inline packages/polymorphic/README.md:86]: Maybe provide a little bit of context as to when/why we'd want to use a default versus not (what value is being provided) -- seeing both examples is confusing without context

File: scripts/utils/tsDocParser.ts (hunk 4 above)
Comment (TheSonOfThomp) [reply scripts/utils/tsDocParser.ts:62]: Yeah, I made these changes to the parser to make sure we weren't publishing docs for the example compnents

**Review by brookescarlett — APPROVED** [review]

File: packages/polymorphic/src/Polymorphic/index.ts (hunk 1 above)
Comment (brookescarlett) [reply packages/polymorphic/src/Polymorphic/index.ts:23]: This needs to be merged in before my PR that converts typography off of box and onto polymorphic
